.gitlab-ci.yml

```diff
@@ -1,10 +1,19 @@
 variables:
   CI: "true"
   DOCKER_DRIVER: overlay2
-  SBT_OPTS: "-Dsbt.ivy.home=$CI_PROJECT_DIR/sbt-cache/ivy"
+  # Make sure sbt cache are captured completely
+  COURSIER_CACHE: $CI_PROJECT_DIR/sbt-cache/coursier
+  SBT_OPTS: >
+    -Dsbt.global.base=$CI_PROJECT_DIR/sbt-cache/sbtboot
+    -Dsbt.boot.directory=$CI_PROJECT_DIR/sbt-cache/boot
+    -Dsbt.ivy.home=$CI_PROJECT_DIR/sbt-cache/ivy
+  # If you changed dependencies (especially for scala), you might want to prevent your branch from using old cache by increase the version blew
+  # gitlab actually use the same technique when you click the `Clear Cache Button` 
+  CACHE_VERSION: 2
```

Comment:
it's actually also reset yarn cache as well (i.e. all cache) 
Put the comments there because scala is more likely to have issues due to how SBT works.
I think we more likely want to reset all cache rather than having separate switches (e.g. SCALA_CACHE_VERSION, PYTHON_CACHE_VERSION & NODE_CACHE_VERSION) to avoid any troubles (so everything will have a fresh start).
Having said that, it's possible to put in the separate cache key. 
Let me know which way you prefer 👍